includes/admin/class-wc-rest-payments-payment-intents-controller.php

```diff
@@ -46,4 +56,35 @@ public function get_payment_intent( $request ) {
 
 		return $this->forward_request( 'get_intent', [ $payment_intent_id ] );
 	}
+
+	/**
+	 * Create a payment intent.
+	 *
+	 * @param WP_REST_Request $request data about the request.
+	 */
+	public function post_payment_intent( $request ) {
+		try {
+			$metadata = $request->get_param( 'metadata' );
+			$order_id = $metadata['order_number'];
+			$order    = wc_get_order( $order_id );
+
+			$wcpay_server_request = Create_Intention::create();
+			$currency             = strtolower( $order->get_currency() );
+			$amount               = WC_Payments_Utils::prepare_amount( $order->get_total(), $currency );
+			$wcpay_server_request->set_currency_code( $currency );
+			$wcpay_server_request->set_amount( $amount );
+			$wcpay_server_request->set_metadata( $metadata );
+			$wcpay_server_request->set_customer( $request->get_param( 'customer' ) );
+			$wcpay_server_request->set_level3( $request->get_param( 'level3' ) );
```

Comment:
`metadata` and `level3` params are not necessary. We should build up them from what we have already had (order_id and order object): 

https://github.com/Automattic/woocommerce-payments/blob/232acdfa2ce16abe1bee8efbd86e31e143909c5e/includes/class-wc-payment-gateway-wcpay.php#L1115-L1116

https://github.com/Automattic/woocommerce-payments/blob/232acdfa2ce16abe1bee8efbd86e31e143909c5e/includes/class-wc-payment-gateway-wcpay.php#L2918-L2921

Reply:
Thanks for pointing me to building metadata and level3 from order data. It reduces the code to call the API on the PoC side as well 👍